src/containers/profile/EditProfile.js

```diff
@@ -0,0 +1,146 @@
+
+/* eslint-disable no-shadow */
+/* eslint-disable camelcase */
+import React, { Component } from 'react';
+import { connect } from 'react-redux';
+import PropType from 'prop-types';
+import { editUserProfile, fetchProfile } from '../../actions/profileActions';
+import firebase from '../../firebase/firebase.config';
+import ProfileEditComponent from '../../components/profile/ProfileEdit';
+
+
+export class ProfileIsEdited extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      firstName: '',
+      lastName: '',
+      dateOfBirth: '',
+      bio: '',
+      image: '',
+      buttonLoading: false,
+    };
+  }
+
+  componentDidMount() {
+    const { fetchProfile, profile: { profile } } = this.props;
+    console.log(profile, 'this is the data from backend');
+    fetchProfile();
+  }
+
+
+  componentWillReceiveProps(nextProps) {
+    console.log(nextProps.profile.profile, 'this is next props');
+    const {
+      first_name, last_name, username, date_of_birth, bio, image,
+    } = nextProps.profile.profile;
+    if (username) {
+      this.setState({
+        firstName: first_name,
+        lastName: last_name,
+        username,
+        image,
+        dateOfBirth: date_of_birth,
+        bio,
+      });
+    }
+  }
+
+  fetchData = () => {
+    this.setState({ buttonLoading: true });
+
+    setTimeout(() => {
+      this.setState({ buttonLoading: false });
+    }, 5000);
+  };
+
+  handleChange = (event) => {
+    event.preventDefault();
+    this.setState({ [event.target.name]: event.target.value });
+  }
+
+  handleSubmit = (event) => {
+    event.preventDefault();
+    this.fetchData();
+    console.log(this.state);
+    const {
+      firstName, lastName, username, dateOfBirth, bio, image,
+    } = this.state;
+
+    const ProfileData = {
+      first_name: firstName,
+      last_name: lastName,
+      username,
+      date_of_birth: dateOfBirth,
+      bio,
+      image,
+    };
+    console.log(ProfileData, 'PROFILEDATA');
+    const { editUserProfile } = this.props;
+    editUserProfile(ProfileData, this.props);
+  }
+
+  /* istanbul ignore next */
+  uploadImage(files) {
+    console.log(files, 'FILES');
+    const fileload = firebase
+      .storage()
+      .ref(`images/${files[0].name}`)
+      .put(files[0]);
+    fileload.then(() => {
+      firebase
+        .storage()
+        .ref(`images/${files[0].name}`)
+        .getDownloadURL()
+        .then((url) => {
+          const image = {
+            image: url,
+          };
+          this.setState({
+            image: image.image,
+          });
+          console.log(image, 'IMAGE');
+        });
+    });
+  }
+
+
+  render() {
+    const {
+      firstName, lastName, dateOfBirth, bio, image, buttonLoading,
+    } = this.state;
+    console.log('STATE.......', image);
```

Comment:
@NicholusMuwonge , kindly remove these console log functions from your code

Reply:
I have reacted to this please